src/main/java/org/networkcalculus/dnc/network/server_graph/ServerGraph.java

```diff
@@ -749,6 +757,38 @@ public Flow getFlow(int id) throws Exception {
 		return result;
 	}
 
+	/**
+	 * Finds all the flows that have exactly path p as their path.
+	 *
+	 * @param p
+	 *             The path which all the returned flows have.
+	 * @return Set of flows that have p as their path.
+	 * @throws Exception
+	 *             If the path is null.
+	 */
+	public Set<Flow> getFlowsByPath(Path p) throws Exception {
+
+        // throw a runtime exception if p is null
+	    if(p == null)
+        {
+            throw new RuntimeException("Path argument for method getFlowsByPath(Path p) in ServerGraph was null.");
+        }
+
+	    else{
+            Set<Flow> result = new HashSet<Flow>();
+            // all the flows that start at first server of the path p are potential candidates
+            Set<Flow> flows_source_p = getFlows(p.getSource());
+            for(Flow f : flows_source_p)
+            {
+                if(f.getPath().equals(p))
+                {
+                    result.add(f);
+                }
+            }
+            return result;
+        }
+	}
```

Comment:
Indentation seems wrong here, too. Probably a mix of tabs and spaces. There's no formatting rules for DNC code (yet) but consistency within methods is appreciated.

Reply:
Looks normal in my IDE though

Reply:
An elusive observation, unfortunately. Hitting tab might result in about four spaces that, in turn, look like a single tab in the IDE.